src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmRpoServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.darts.arm.service.impl;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.arm.service.ArmRpoService;
+import uk.gov.hmcts.darts.authorisation.component.UserIdentity;
+import uk.gov.hmcts.darts.common.entity.ArmRpoExecutionDetailEntity;
+import uk.gov.hmcts.darts.common.entity.ArmRpoStatusEntity;
+import uk.gov.hmcts.darts.common.exception.DartsException;
+import uk.gov.hmcts.darts.common.helper.CurrentTimeHelper;
+import uk.gov.hmcts.darts.common.repository.ArmRpoExecutionDetailRepository;
+
+@Service
+@AllArgsConstructor
+public class ArmRpoServiceImpl implements ArmRpoService {
+
+    public static final String ARM_RPO_EXECUTION_DETAIL_NOT_FOUND = "ArmRpoExecutionDetail not found";
+    private final CurrentTimeHelper currentTimeHelper;
+    private final UserIdentity userIdentity;
+    private final ArmRpoExecutionDetailRepository armRpoExecutionDetailRepository;
+
+    @Override
+    public ArmRpoExecutionDetailEntity getArmRpoExecutionDetailEntity(Integer executionId) {
+        return armRpoExecutionDetailRepository.findById(executionId).orElseThrow(() -> new DartsException(ARM_RPO_EXECUTION_DETAIL_NOT_FOUND));
```

Comment:
Should have a *ApiErrror code apart of the exception

Reply:
These are all called from automated jobs not the main application so thats why it returns the DartsException instead.